src/functions/getNextUploadBatch/framework/handler.ts

```diff
@@ -0,0 +1,53 @@
+import { APIGatewayEvent, Context } from 'aws-lambda';
+import { getNextUploadBatch } from '../application/next-update-batch-service';
+import createResponse from '../../../common/application/utils/createResponse';
+import { HttpStatus } from '../../../common/application/api/HttpStatus';
+import Response from '../../../common/application/api/Response';
+import { InterfaceTypes } from '../domain/interface-types';
+import { BatchRequestParamaterErrors } from '../domain/batch-request-param-errors';
+import { gzipSync } from 'zlib';
+
+export async function handler(event: APIGatewayEvent, fnCtx: Context): Promise<Response> {
+  let nextBatchData;
+  const interfaceType = convertToInterfaceType(event.queryStringParameters.interface);
+  const batchSize = event.queryStringParameters.batch_size;
+  const validationResponse = validateQueryParameters(interfaceType, batchSize);
+
+  if (validationResponse !== undefined) {
+    return createResponse(validationResponse, HttpStatus.BAD_REQUEST);
```

Comment:
Could bring in a simple joi schema here to validate types on the incoming event, then we could return a useful error message to the caller.

Reply:
Yes, that actually does sound good. Let me have a play around with it.